### DISCUSSION TITLE
ag.scheduler.HyperbandScheduler seems to be gone in 4.0.0 (it was there in autogluon 0.0.15)

### DISCUSSION THREAD

**Discussion by danielkorzekwa** [autogluon/autogluon#1649] [Q&A] [answered]
I was doing Hyper-Parameter Optimization (HPO) using Autogluon 0.0.15, following a nice wiki: https://auto.gluon.ai/scoredebugweight/tutorials/course/algorithm.html and using the ag.scheduler.HyperbandScheduler(...)

This functionality is not available anymore in AutoGluon 4.0.0. What happened?

Is Autogluon still aiming to be a generic framework for HPO optimization?

**Comment by Innixma** [comment]
We are currently refactoring our HPO functionality to be more scalable and maintainable. v0.4 is currently in a transition process from the old logic to the new logic, and the limited scope of HPO in v0.4 was required to enable Windows OS support and easier installation. If you are interested in doing hyperband HPO, you may want to use v0.3.1 which should still have the functionality. More HPO functionality will be reintroduced in v0.5, although the main focus of AG is on AutoML and not necessarily HPO.

**Comment by danielkorzekwa** [comment, marked as answer]
Yes, makes sense now. 

I would clarify on the project homepage/wikis that the current v0.4 is in a transition process, and put some timelines for the future project iterations. Otherwise you push potential users away from using Autogluon. Current documentation for v0.4 creates impression that the quality standards in Autogluon are poor, which does not encourage users to think that they can rely on Autogluon in their systems.
